packages/opencensus-exporter-trace-zipkin/src/zipkin.ts

```diff
@@ -0,0 +1,178 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as http from 'http';
+import * as url from 'url';
+
+export interface ZipkinExporterOptions extends types.ExporterConfig {
+  url: string;
+  serviceName: string;
+}
+
+interface TranslatedSpan {
+  traceId: string;
+  name: string;
+  id: string;
+  parentId?: string;
+  kind: string;
+  timestamp: string;
+  duration: string;
+  debug: boolean;
+  shared: boolean;
+  localEndpoint: {serviceName: string};
+}
+
+/** Zipkin Exporter manager class */
+export class ZipkinTraceExporter implements types.Exporter {
+  private zipkinUrl: url.UrlWithStringQuery;
+  private serviceName: string;
+  buffer: classes.ExporterBuffer;
+  logger: types.Logger;
+
+  constructor(options: ZipkinExporterOptions) {
+    this.zipkinUrl = url.parse(options.url);
+    this.serviceName = options.serviceName;
+    this.buffer = new classes.ExporterBuffer(this, options);
+    this.logger = options.logger || logger.logger();
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.buffer.addToBuffer(root);
+  }
+
+  /**
+   * Send a trace to zipkin service
+   * @param zipkinTraces Trace translated to Zipkin Service
+   */
+  private async sendTraces(zipkinTraces: TranslatedSpan[]) {
+    /** Request options */
+    const options = {
+      hostname: this.zipkinUrl.hostname,
+      port: this.zipkinUrl.port,
+      path: this.zipkinUrl.path,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    };
+
+    return new Promise((resolve, reject) => {
+      try {
+        /** Request object */
+        const req = http.request(options, (res) => {
+          res.on('data', (chunk) => {});
+          // Resolve on end
+          res.on('end', () => {
+            resolve(
+                {statusCode: res.statusCode, statusMessage: res.statusMessage});
+          });
+        });
+
+        /** Request error event */
+        req.on('error', (e) => {
+          reject({
+            statusCode: 500,
+            statusMessage: `Problem with request: ${e.message}`
+          });
+        });
+
+        /** Request body */
+        const spansJson: string[] =
+            zipkinTraces.map((span) => JSON.stringify(span));
+        spansJson.join('');
```

Comment:
Should this be `spansJson.join(',');`? Or maybe just `const outputJson = JSON.stringify(zipkinTraces)`?